src/address/types.ts

```diff
@@ -23,10 +23,11 @@ export interface Address {
 }
 
 export interface GetAddressResponse {
-  addresses: Array<Address>
+  addresses: Array<Address>;
 }
 
 export interface GetAddressOptions {
+  getProvider?: () => Promise<BitcoinProvider>;
```

Comment:
The type `() => Promise<BitcoinProvider>` is repeated in quite a few places. It should also be coming up with a Typescript error in most of them since `getDefaultProvider` doesn't return a Promise 🤔 

Could you please add a new central type in `src/provider/index.ts` for this which we can reference throughout the repo? It should also have the following structure:
`type GetBitcoinProviderFunc = () => Promise<BitcoinProvider | undefined> | BitcoinProvider | undefined`

Reply:
Done in c1f1a8e.

I did, however, keep the function as an async function to simplify the type. If you would prefer to relax the type, let me know and I can do the change.

Reply:
That looks good 👍 The code itself can handle a non-async function anyways, so if someone isn't being type safe, it'll still work, so all good.